apps/functions-subscription/src/use-cases/__tests__/add-subscription.test.ts

```diff
@@ -0,0 +1,83 @@
+import { describe, expect, it } from 'vitest';
+import * as IO from 'fp-ts/IO';
+import * as TE from 'fp-ts/TaskEither';
+import * as O from 'fp-ts/Option';
+import * as E from 'fp-ts/Either';
+import { insertSubscription } from '../insert-subscription';
+import {
+  aSubscription,
+  aSubscriptionRequest,
+} from '../../domain/__tests__/data';
+import { makeTestEnv } from '../../domain/__tests__/mocks';
+import { SubscriptionAlreadyExists } from '../errors';
+
+describe('insertSubscription', () => {
+  it('should return error if the subscription already exists', async () => {
+    const { userId, trialId } = aSubscription;
```

Comment:
**suggestion (non-blocking)**: you can avoid to extract `userId` and `trialId` in every test

Since the `userId` and the `trialId` are constants, and the `aSubscription` should be immutable, you can refactor the tests in this way:

```
describe('insertSubscription', () => {
    const { userId, trialId } = aSubscription;

    it('should return error if the subscription already exists', async () => {// you can use here `userId` and `trialId`});

    t('should return the subscription created', async () => {// you can use here `userId` and `trialId`});
});
```

Reply:
Yes right! Done ✅